src/inflators/spot-light.ts

```diff
@@ -0,0 +1,53 @@
+import { addComponent } from "bitecs";
+import { addObject3DComponent } from "../utils/jsx-entity";
+import { SpotLightTag, LightTag } from "../bit-components";
+import { SpotLight } from "three";
+import { HubsWorld } from "../app";
+
+export type SpotLightParams = {
+  color: string;
+  intensity: number;
+  range: number;
+  decay: number;
+  innerConeAngle: number;
+  outerConeAngle: number;
+  castShadow: boolean;
+  shadowMapResolution: [number, number];
```

Comment:
Not particularly important, but you can name the elements of a "tuple". Gives you slightly nicer type hinting:

```ts
  shadowMapResolution: [width: number, height: number];
```